examples/apps/generic_client/base_generic_client.py

```diff
@@ -0,0 +1,152 @@
+from avalon_sdk.work_order.work_order_params import WorkOrderParams
+import avalon_crypto_utils.signature as signature
+import avalon_crypto_utils.crypto_utility as crypto_utility
+import verify_report.verify_attestation_report as attestation_util
+from error_code.error_status import SignatureStatus
+
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO)
+
+
+class BaseGenericClient(GenericClientInterface):
+    """
+    Abstract class for generic client
+    It has common functions which are common to both
+    direct and proxy model.
+    """
+    def __init__(self):
+        super().__init__()
+        self._worker_registry_list_instance = None
+        self._worker_instance = None
+        self._work_order_instance = None
+        self._work_order_receipt_instance = None
+
+    def do_worker_verification(self, worker_obj):
+        """
+        Do worker verification on proof data if it exists
+        Proof data exists in SGX hardware mode.
+        """
+        # TODO Need to do verify MRENCLAVE value
+        # in the attestation report
+        if not worker_obj.proof_data:
+            logging.info("Proof data is empty. " +
+                         "Skipping verification of attestation report")
+        else:
+            # Construct enclave sign-up info json
+            enclave_info = {
+                'verifying_key': worker_obj.verification_key,
+                'encryption_key': worker_obj.encryption_key,
+                'proof_data': worker_obj.proof_data,
+                'enclave_persistent_id': ''
+            }
+
+            logging.info("Perform verification of attestation report")
+            verify_report_status = attestation_util.verify_attestation_report(
+                enclave_info)
+            if verify_report_status is False:
+                logging.error("Verification of enclave sign-up info failed")
+                return False
+            else:
+                logging.info("Verification of enclave sign-up info passed")
+                return True
+
+    def create_work_order_params(self, worker_id, workload_id,
+                                 in_data, worker_encrypt_key,
+                                 session_key, session_iv,
+                                 enc_data_enc_key):
+        """
+        Create work order request params
+        """
+        work_load_id = workload_id.encode("UTF-8").hex()
+        self._work_order_id = secrets.token_hex(32)
+        self._session_key = session_key
+        self._session_iv = session_iv
+        requester_id = secrets.token_hex(32)
+        requester_nonce = secrets.token_hex(16)
+        # Create work order params
+        try:
+            wo_params = WorkOrderParams(
+                self._work_order_id, worker_id, work_load_id, requester_id,
+                self._session_key, self._session_iv, requester_nonce,
+                result_uri=" ", notify_uri=" ",
```

Comment:
are result_uri & notify_uri deliberately having a space ?

Reply:
These URIs are not being used and they are optional parameters. we can skip them in constructor.